File: src/lua/alu/quicksort.lua
QuickSort = class(CrystalBenchmarkBase, function(quickSort, writer)
    CrystalBenchmarkBase.init(quickSort, writer, "QuickSort")
end)

QuickSort.MemSize = 2048

function QuickSort:benchImplementation()
    local count = 0
    local temp = QuickSort.MemSize / 4
    local holdrand = 1
    local memory = {}
    local test = {}

    for i = 1, temp, 1 do
        holdrand = bitand(rshift((holdrand * 214013 + 2531011), 16), 0x7fff)
        memory[i] = holdrand
        test[i] = 0
    end

    while not self:isCancelled() do
        for i = 1, Eratosthenes.N, 1 do test[i] = memory[i] end

        table.sort(test)

        count = count + 1
    end

    return count / 4.7 / 25.0
end

File: src/dotnet/EntityFX.CrystalMarkNet/ALU/QuickSort.cs
using System;
using System.Threading;

namespace EntityFX.CrystalMarkNet.ALU
{
    class QuickSort : CrystalBenchmarkBase
    {
        private const int MemSize = 2048;

        protected override double BenchImplementation(CancellationToken cancellationToken)
        {
            uint[] memory = new uint[MemSize / 4];
            uint[] test = new uint[MemSize / 4];
            uint count = 0;
            int temp = MemSize / 4;

            uint holdrand = 1;
            for (int i = 0; i < temp; i++)
            {
                holdrand = (uint)((holdrand * 214013L + 2531011L) >> 16) & 0x7fff;

                memory[i] = holdrand;
            }

            for (; ; )
            {
                for (int i = 0; i < temp; i += 4)
                {
                    test[i] = memory[i];
                    test[i + 1] = memory[i + 1];
                    test[i + 2] = memory[i + 2];
                    test[i + 3] = memory[i + 3];
                }
                //Array.Copy(memory, test, temp);
                Array.Sort(test);
                count++;

                if (cancellationToken.IsCancellationRequested)
                {
                    return count / 4.7d / 5.0d;
                }
            }
        }
    }
}